Generate RDF from ResourceTemplateForm

jermnelson opened the pull request:
Starts support for generating a simple RDF graph for components and props in a `ResourceTemplateForm`.  

- New reducer generates RDF based on the based on the Redux stores for `InputList`, `InputLiteral`, and `InputLookup`. 
- Adds a new key, **rtId** for storing the `ResourceTemplate` id in `InputList`, `InputLiteral`, and `InputLookup`, this is used to scope the generation of triples to a single ResourceTemplate
- Uses [N3](https://github.com/rdfjs/N3.js) JS module for RDF manipulation. 
- New Preview RDF button displays a JS alert with the serialized graph in Turtle format. 
- All subjects are blank nodes
### New Preview RDF Button
<img width="744" alt="screen shot 2019-01-11 at 12 25 41 pm" src="https://user-images.githubusercontent.com/71847/51055096-0edda900-159c-11e9-8f23-efa7f12bc2c6.png">

### Alert with RDF Serialization in Turtle
<img width="557" alt="screen shot 2019-01-11 at 9 57 47 am" src="https://user-images.githubusercontent.com/71847/51055113-20bf4c00-159c-11e9-83a1-b15eba6dff7e.png">

## Know Issues and Future Work
- Only works for `ResourceTemplateForm`s that are **NOT** embedded in a modal; issue is ticketed in #281. 
- Once #281 is completed, will need to return the blank node or uri to the calling `ResourceTemplateForm` to complete the triple, ticketed in #282
- Should use the N3 Store instead of a N3 writer for adding triples to the graphs. Using `N3.Store` may make it easier to send RDF back to Trellis in the Sinopia Server. Ticket #280 
- Need to either mint a new URI from a Sinopia Server call or create a dummy URI for client-side RDF subjects for those `ResourceTemplateForm`s that need a URI for the subject. Ticket #283

jermnelson commented:
Thanks @SaravShah! I don't think the InputLookup and InputList needs the language selection because they actually are just URI references in the RDF to external resources. Those sources would have the actual literals for such things as labels that may have the language attribute set.

jgreben commented:
When I run this branch locally I get some console errors:

`checkPropTypes.js:19 Warning: Failed prop type: Invalid prop `literals` of type `object` supplied to `ResourceTemplateForm`, expected `array`.`

`checkPropTypes.js:19 Warning: Failed prop type: Invalid prop `lookups` of type `object` supplied to `ResourceTemplateForm`, expected `array`.`

`checkPropTypes.js:19 Warning: Failed prop type: The prop `rtType` is marked as required in `ResourceTemplateModal`, but its value is `undefined`.`

Also, the RDF for the LOC names lookup does not seem right:
![screen shot 2019-01-10 at 9 48 54 am](https://user-images.githubusercontent.com/3093850/50986990-28f48a00-14bd-11e9-929b-518c3384e8fc.png)

ndushay commented:
actually, would be awesome to see images of button added and what clicking the button looks like.  I use "command - control - shift - f4" to take cropped screenshot and then I paste it into the description.

I found this practice super helpful when last I worked with folks on access team.

jermnelson commented:
Hi @jgreben, latest commit fixes prop types errors and adjusted `InputLookup` to use propertyURI instead of propertyLabel for the **id** which fixes  the RDF for the **LOC names lookup** above.

Good idea about the screenshots @ndushay, added screen shots to the PR description.

jgreben commented on src/reducers/rdf.js:
```diff
@@ -0,0 +1,72 @@
+  graph: {}
+}
+
+const addTriples = (subject, formData, rtId, testProperty, graph) => {
```
I wouldn't mind seeing if we can refactor this so that we are not always passing in all these arguments. Maybe separate functions to handle 'uri' and 'content'/other (e.g. addTriplesForUri())? I'd be happy to work on this as part of a follow-on PR. Non-blocker for this PR.

jermnelson replied:
Yeah, not sure what the best option is here; originally I had this functionality within the `generateMyRDF` with two forEach loops.
